pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -441,19 +439,46 @@ func (r *ReconcileWindowsMachine) isAllowedDeletion(machine *mapi.Machine) bool
 		return false
 	}
 
-	if minHealthyCount == *(windowsMachineSet.Spec.Replicas) {
+	// Allow deletion if there is only one machine in the Windows MachineSet
+	totalWindowsMachineCount := *windowsMachineSet.Spec.Replicas
+	if maxUnhealthyCount == totalWindowsMachineCount {
 		return true
 	}
 
 	totalHealthy := 0
 	for _, ma := range machines.Items {
-		// Machines are determined as Healthy when they are part of Windows MachineSet and are
-		// in the Running Status
-		if len(ma.OwnerReferences) != 0 && ma.OwnerReferences[0].Name == machinesetName {
-			if ma.Status.Phase != nil && *ma.Status.Phase == "Running" && ma.Status.NodeRef != nil {
-				totalHealthy += 1
-			}
+		// Increment the count if the machine is identified as healthy and is a part of given Windows MachineSet and
+		// on which deletion is not already initiated.
+		if r.isWindowsMachineHealthy(&ma) && ma.OwnerReferences[0].Name == machinesetName &&
+			ma.DeletionTimestamp.IsZero() {
+			totalHealthy += 1
 		}
 	}
-	return totalHealthy-1 >= minHealthyCount
+
+	return totalWindowsMachineCount-int32(totalHealthy) < maxUnhealthyCount
+}
+
+// isWindowsMachineHealthy determines if the given Machine object is healthy. A Windows machine is considered
+// unhealthy if -
+// 1. Machine is not mapped to a Windows MachineSet
+// 2. Machine is not in a 'Running' phase
+// 3. Machine is not associated with a Node object
+// 4. Associated Node object doesn't have a Version annotation
+func (r *ReconcileWindowsMachine) isWindowsMachineHealthy(machine *mapi.Machine) bool {
+	if len(machine.OwnerReferences) == 0 && (machine.Status.Phase == nil || *machine.Status.Phase != "Running") &&
```

Comment:
I understand why you are making this check `len(machine.OwnerReferences) == 0`, but I'm wondering if this belongs within this function.
Machines can be created separate of a MachineSet. We don't support it, but technically can't that be a healthy Machine?
You already make sure the machine has an owner before you call the function.